### PR TITLE
Android Embedding PR 11: Add FlutterEngine to FlutterFragment.

### PR DESCRIPTION
**Pull request by matthew-carroll**

### CODE REVIEW

File: shell/platform/android/io/flutter/embedding/engine/android/FlutterFragment.java [hunk 1]
```diff
@@ -144,9 +149,62 @@ protected static Bundle createArgsBundle(@Nullable String dartEntrypoint,
...
+    onFlutterEngineCreated(flutterEngine);
+  }
+
+  /**
+   * Hook for subclasses to customize the creation of the {@code FlutterEngine}.
+   *
+   * By default, this method returns a standard {@link FlutterEngine} without any modification.
+   */
+  @NonNull
+  protected FlutterEngine onCreateFlutterEngine(@NonNull Context context) {
```
Comment (dnfield) [inline shell/platform/android/io/flutter/embedding/engine/android/FlutterFragment.java]: The naming of this seems awkward to me.  I think this should just be called `createFlutterEngine`.  Having a non-`void` method called `onCreate` seems unusual.

Comment (dnfield) [reply shell/platform/android/io/flutter/embedding/engine/android/FlutterFragment.java]: This is really a nit though, feel free to disregard if it fits some other pattern that I'm just not aware of.

Comment (matthew-carroll) [reply shell/platform/android/io/flutter/embedding/engine/android/FlutterFragment.java]: The convention I was trying to follow here is similar to the `onCreateView()` method in Fragments.  There's a primary method invoked in the superclass, in this case called `createFlutterEngine()`, which then internally invokes a subclass hook, in this case called `onCreateFlutterEngine()`.  Does that make sense?

Comment (dnfield) [reply shell/platform/android/io/flutter/embedding/engine/android/FlutterFragment.java]: Fair enough

**Pull request merged**